pin/pin.go

```diff
@@ -97,16 +102,19 @@ func (p *pinner) Pin(node *mdag.Node, recurse bool) error {
 			return err
 		}
 	} else {
+		if p.recursePin.HasKey(k) {
+			return errors.New("Key already pinned recursively.")
+		}
 		p.directPin.AddBlock(k)
 	}
 	return nil
 }
 
-// Unpin a given key with optional recursive unpinning
-func (p *pinner) Unpin(k util.Key, recurse bool) error {
+// Unpin a given key
+func (p *pinner) Unpin(k util.Key) error {
 	p.lock.Lock()
 	defer p.lock.Unlock()
-	if recurse {
+	if p.recursePin.HasKey(k) {
```

Comment:
It is still unclear. `rm dir/` does not remove directories without `-r`. I think this is a safe practice from users accidentally deleting a lot of stuff they didn't mean to. pinning is one of those dangerous things we must exercise extra caution with.